### DISCUSSION TITLE
Osclass v5.1.0

### DISCUSSION THREAD

**Discussion by navjottomer** [mindstellar/shopclass#435] [General]
## Updated changelog for Osclass 5.1.0
* New: Backend is based on Bootstrap 5
* New: Many changes are made to make it more user-friendly on small screens while keeping the same functionality
* New: System info page in the tools menu.
* New: form input class is introduced, which will be the new default for all form elements, soon a new API will be introduced to use it.
* New: Translations, from mindstellar/i10n-osclass
* New: Versions of translation can be downloaded from our repositories in the Osclass backend
* New: Language has info for text direction, which can be utilized by developers
* New: Translations for new languages
* New: Translated custom fields. They can be translated into the custom-field editor.
* New: open-in-new-tab option for URL type custom fields.
* New: Login as a user feature. Can be used to log in as another user. You'll find this option on the user edit page. Thanks, @dftd
* New: You can enable the allow-prerelease setting in the user dashboard to get new features and bug fixes and continue to test this version.
* Improvement: In osclass upgrade experience
* Improvement: Less dependency on JQuery-UI and other libraries, the goal is to remove them completely with native javascript or use bootstrap components
* Improvement: Major rewrite of old Jquery based code to Pure JS (Still lots to do)
* Improvement: Escape, Sanitize classes, and they are used now in many places
* Security: Many security flaws are fixed, thanks to new classes
* Improvement: Locations stats generation, on average a 10 x improvement in performance
* Improvement: Compatibility with older plugins but not for ones that use ancient DB access methods.
* Improvement: Huge reductions in PHP Notices and warnings with newer PHP versions
* Improvement: Now styles can be registered like you have registered scripts.
* Improvement: JS scripts can be enqueued in the middle of view and will be executed at the end of the page.
* Speed: Overall performance improvements and code refactoring to make it more stable and easier to maintain
* Speed: Database improvements, from 10x to 5x faster in some cases, reduced the query footprint in many cases
* Changed: Default DB engine from MyISAM to InnoDB for t_item_description table.
* Fixed: JS enqueued in the footer was not working in some cases.
* Fixed: Multilingual issue with the backend.
* Fixed: #400: Issues storing ipv6 addresses in a few places.
* Fixed: Fresh installation of Osclass does not work.
* Removed: the google-map plugin. as functionality is now provided by the core.
over 300 + commits since the last release
There is a lot more under the hood changes, which are not listed here, but you can see them in the commit log.

Backend rewrite was a lot of work, but now it is finally done, and it is much more user-friendly. Many new components/APIs will be introduced in the future. So, if you are a developer, that'll make you happy. It was a marathon job for the last two to three months, and I am very happy with the result. I hope you will enjoy it.

**If you want to help or donate to support the project? Just buy me little coffee at https://www.paypal.com/paypalme/navjottomer, it will be great :)**

**Full Changelog**: https://github.com/mindstellar/Osclass/compare/5.0.0...5.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/mindstellar/Osclass/releases/tag/5.1.0'>Osclass v5.1.0</a>.</em>

**Comment by buninsan** [comment]
In the new version of the site, the Show e-mail checkbox is always enabled, it cannot be turned off even by the administrator.
When you delete a category in the site admin area, a redirect to the console occurs and this is very inconvenient.
Correct it please.

**Reply by eurobank** [reply]
@buninsan Show email where? You mean the Admin Ad Edit? Because when adding something as user, i don't see any of it.

**Reply by buninsan** [reply]
https://www.dropbox.com/s/a4x1dkqdxx2474j/123Video_2021-11-17_211719.wmv?dl=0

**Reply by eurobank** [reply]
ok, so this is for the Admin dashboard when editing or adding a new ad.

**Reply by buninsan** [reply]
No!
This checkbox is active in all ads, in all!

**Reply by eurobank** [reply]
@buninsan The capture you posted shows the admin dashboard. 

In my sites it doesn't show to any USER adding a classified ads.

Can you post a USER posting an ad capture? What theme is that?

**Reply by buninsan** [reply]
Are you using Osclass 5.1.0?

**Reply by eurobank** [reply]
Yes, 5.1 dev

**Reply by eurobank** [reply]
@navjottomer There is indeed some issue in **Admin EDIT ads,** the Show email is Checked (active) and it should be off.

**Reply by eurobank** [reply]
@buninsan Do you allow NOT registered (logged in) users to post? In that case i see the checkbox.

**Reply by navjottomer** [reply]
> @navjottomer There is indeed some issue in **Admin EDIT ads,** the Show email is Checked (active) and it should be off. 

Okay, I'll check.

**Reply by navjottomer** [reply]
See #441

**Comment by eurobank** [comment]
Osclass 5.1 is actually a very stable and good version. Thank Navjot (and Defected).
